src/Schema/Directives/ScopeDirective.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use BadMethodCallException;
+use Nuwave\Lighthouse\Exceptions\DirectiveException;
+use Nuwave\Lighthouse\Support\Contracts\DefinedDirective;
+use Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective;
+
+class ScopeDirective extends BaseDirective implements ArgBuilderDirective, DefinedDirective
+{
+    /**
+     * Name of the directive.
+     *
+     * @return string
+     */
+    public function name()
+    {
+        return 'scope';
+    }
+
+    public static function definition(): string
+    {
+        return /* @lang GraphQL */ <<<'SDL'
+"""
+This directive adds a scope to the builder and supplies the argument value as the argument to the scope.
+"""
+directive @scope(
+  """
+  The name of the scope.
+  """
+  name: String
+) on ARGUMENT_DEFINITION
+SDL;
+    }
+
+    /**
+     * Add additional constraints to the builder based on the given argument value.
+     *
+     * @param \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder $builder
+     * @param mixed $value
+     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+     * @throws DirectiveException
+     */
+    public function handleBuilder($builder, $value)
+    {
+        $scope = $this->directiveArgValue('name');
+        try {
+            return $builder->{$scope}($value);
+        } catch (BadMethodCallException $exception) {
+            throw new DirectiveException(
```

Comment:
I am planning to phase that out in favour of the `DefinitionException`, used to universally indicate any kind of schema definition error.